src/components/features/DashboardView.tsx

```diff
@@ -0,0 +1,35 @@
+import { useGetTasksQuery } from "../../gql/graphql";
+import Error from "./../errors/Error";
+import TaskColumnSkeleton from "../ui/loadingSkeletons/TaskColumnSkeleton";
+import { useAppStore } from "../../store/store";
+import GridView from "./GridView";
+import TableView from "./TableView";
+import { statusOptions } from "../../libs/utils";
+
+const DashboardView = () => {
+  const viewMode = useAppStore((state) => state.viewMode);
+  const searchByTaskName = useAppStore((state) => state.searchByTaskName);
+
+  const { loading, error, data } = useGetTasksQuery({
+    variables: searchByTaskName !== ""
+    ? { input: { name: searchByTaskName } }
+    : { input: {} },
+  });
+
+  const tasks = data?.tasks || [];
+
+  if (loading) return <TaskColumnSkeleton />;
+  if (error) return <Error message={error.message} />;
```

Comment:
your error component receives an on retry prop, should we pass a refetch function in that case?